File: ws/pilaguano/u1/ws08CDJavaClasses/ChickenFarmSimulator/src/utils/FileManager.java
package utils;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author David Pilaguano
 */
public class FileManager {

    public static void save(String data, String fileName, String type) throws IOException {
        try (FileWriter writer = new FileWriter(fileName + "." + type, true)) {
            writer.write(data + "\n");
        }
    }

    public static List<String> read(String fileName, String type) throws IOException {
        List<String> data = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(fileName + "." + type))) {
            String line;
            while ((line = reader.readLine()) != null) {
                data.add(line);
            }
        }
        return data;
    }

    public static void update(String oldData, String newData, String fileName, String type) throws IOException {
        List<String> fileContent = read(fileName, type);
        try (FileWriter writer = new FileWriter(fileName + "." + type, false)) {
            for (String line : fileContent) {
                if (line.equals(oldData)) {
                    writer.write(newData + "\n");
                } else {
                    writer.write(line + "\n");
                }
            }
        }
    }

    public static void delete(String data, String fileName, String type) throws IOException {
        List<String> fileContent = read(fileName, type);
        try (FileWriter writer = new FileWriter(fileName + "." + type, false)) {
            for (String line : fileContent) {
                if (!line.equals(data)) {
                    writer.write(line + "\n");
                }
            }
        }
    }

    public static List<String> find(String searchData, String fileName, String type) throws IOException {
        List<String> foundData = new ArrayList<>();
        List<String> fileContent = read(fileName, type);
        for (String line : fileContent) {
            if (line.contains(searchData)) {
                foundData.add(line);
            }
        }
        return foundData;
    }
}

File: ws/pilaguano/u1/ws08CDJavaClasses/ChickenFarmSimulatorPython/Utils/file_manager.py
# utils/file_manager.py

import os
import json

class FileManager:
    @staticmethod
    def save(data, path, extension):
        with open(f"{path}.{extension}", 'w') as file:
            file.write(data)

    @staticmethod
    def find(keyword, path, extension):
        results = []
        with open(f"{path}.{extension}", 'r') as file:
            lines = file.readlines()
            for line in lines:
                if keyword in line:
                    results.append(line)
        return results

    @staticmethod
    def update(old_data, new_data, path, extension):
        with open(f"{path}.{extension}", 'r') as file:
            lines = file.readlines()
        with open(f"{path}.{extension}", 'w') as file:
            for line in lines:
                if old_data in line:
                    file.write(line.replace(old_data, new_data))
                else:
                    file.write(line)

    @staticmethod
    def delete(data_to_delete, path, extension):
        with open(f"{path}.{extension}", 'r') as file:
            lines = file.readlines()
        with open(f"{path}.{extension}", 'w') as file:
            for line in lines:
                if data_to_delete not in line:
                    file.write(line)
